lib/controllers/spree/api/v2/storefront/intents_controller.rb

```diff
@@ -5,6 +5,24 @@ module Storefront
         class IntentsController < ::Spree::Api::V2::BaseController
           include Spree::Api::V2::Storefront::OrderConcern
 
+          def payment_confirmation_data
+            spree_authorize! :update, spree_current_order, order_token
+
+            if spree_current_order.intents?
+              spree_current_order.process_payments!
+              spree_current_order.reload
+              last_valid_payment = spree_current_order.payments.valid.where.not(intent_client_key: nil).last
+
+              if last_valid_payment.present?
+                client_secret = last_valid_payment.intent_client_key
+                publishable_key = last_valid_payment.payment_method.preferred_publishable_key
+                return render json: { client_secret: client_secret, pk_key: publishable_key }, status: :ok
+              end
+            end
+
+            render json: { client_secret: nil, pk_key: nil }, status: :ok
```

Comment:
@rafalcymerys shouldn't we return 422 here?

Reply:
That's a good question. My thinking was that it's a correct state, as this can happen for example when a given payment method doesn't have intents enabled. In that case I didn't want to return an error code, but rather a response that there's no remaining action here.

Any thoughts on that?

Reply:
I think that could lead to false positives and be hard to debug... Returning 422 and logging it via rails logger will enable to answer a lot of questions in the future ;)